includes/core/server/request/class-create-and-confirm-intention.php

```diff
@@ -0,0 +1,217 @@
+				'wcpay_core_invalid_request_parameter_metadata_order_number'
+			);
+		}
+
+		// The description is based on the order number here.
+		$description = $this->get_intent_description( $metadata['order_number'] ?? 0 );
+		$this->set_param( 'description', $description );
+
+		// Combine the metadata with the fingerprint.
+		$metadata = array_merge( $metadata, $this->get_fingerprint_metadata() );
+		$this->set_param( 'metadata', $metadata );
+
+		return $this;
+	}
+
+	/**
+	 * Level 3 data setter.
+	 *
+	 * @param  array $level3 Level 3 data.
+	 * @return static        Instance of the class for method chaining.
+	 */
+	public function set_level3( $level3 ) {
+		if ( empty( $level3 ) || ! is_array( $level3 ) ) {
+			return $this;
+		}
+
+		return $this->set_param( 'level3', $this->fix_level3_data( $level3 ) );
+	}
+
+	/**
+	 * Off-session setter.
+	 *
+	 * @param  bool $off_session Whether the payment is off-session (merchant-initiated), or on-session (customer-initiated).
+	 * @return static            Instance of the class for method chaining.
+	 */
+	public function set_off_session( bool $off_session = true ) {
+		// This one is tricky. We can have `true`, but otherwise we need to get rid of the parameter.
+		if ( $off_session ) {
+			return $this->set_param( 'off_session', true );
+		} else {
+			return $this->unset_param( 'off_session' );
+		}
+	}
+
+	/**
+	 * Payment methods setter.
+	 *
+	 * @param  array $payment_methods               An array of payment methods that might be used for the payment.
+	 * @return static                               Instance of the class for method chaining.
+	 * @throws Invalid_Request_Parameter_Exception  When there are no payment methods provided.
+	 */
+	public function set_payment_methods( array $payment_methods ) {
+		// Hard to validate without hardcoding a list here.
+		if ( empty( $payment_methods ) ) {
+			throw new Invalid_Request_Parameter_Exception(
+				__( 'Intentions require at least one payment method', 'woocommerce-payments' ),
+				'wcpay_core_invalid_request_parameter_missing_payment_method_types'
+			);
+		}
+
+		return $this->set_param( 'payment_methods_types', $payment_methods );
+	}
+
+	/**
+	 * CVC confirmation setter.
+	 *
+	 * @param  string $cvc_confirmation The CVC confirmation for this payment method (Optional).
+	 * @return static                   Instance of the class for method chaining.
+	 */
+	public function set_cvc_confirmation( $cvc_confirmation = null ) {
+		return $this->set_param( 'cvc_confirmation', $cvc_confirmation );
+	}
+
+	/**
+	 * Formats the response from the server.
+	 *
+	 * @param  mixed $intention_array The response from `WC_Payments_API_Client::request`.
+	 * @return mixed                  Either the same response, or the correct object.
+	 */
+	public function format_response( $intention_array ) {
+		return WC_Payments::get_payments_api_client()->deserialize_intention_object_from_array( $intention_array );
```

Comment:
For consistency and future-proof 
```suggestion
return new Response( WC_Payments::get_payments_api_client()->deserialize_intention_object_from_array( $intention_array ) );
```